### PR TITLE
Refactor/shared preferences

### LINKED ISSUE #609: Make sharedPreferences instance static.
Maybe as a followup and but once rebased I'd love to also remove the sharedPrefs from here.

Especially as we use the same one throughout the app a static field somewhere would also work (maybe in the NeonPlatform).

_Originally posted by @Leptopoda in https://github.com/nextcloud/neon/pull/495#discussion_r1306648141_

### PR DESCRIPTION
**Pull request by Leptopoda**
fixes #609

### CODE REVIEW

File: packages/app/lib/apps.dart [hunk 1]
```diff
@@ -3,14 +3,10 @@ import 'package:neon_files/neon_files.dart';
...
-List<AppImplementation> getAppImplementations(
-  final SharedPreferences sharedPreferences,
-) =>
-    [
-      FilesApp(sharedPreferences),
-      NewsApp(sharedPreferences),
-      NotesApp(sharedPreferences),
-      NotificationsApp(sharedPreferences),
-    ];
+final List<AppImplementation> appImplementations = [
```
Comment (provokateurin) [inline packages/app/lib/apps.dart:7]: Since there is no logic left I would just inline this

**Review by provokateurin — APPROVED** [review]
Just one small nit pick, very nice cleanup!

File: packages/app/lib/apps.dart (hunk 1 above)
Comment (Leptopoda) [reply packages/app/lib/apps.dart:7]: This isn't possible as we still need it for the screenshot tests

**Review by provokateurin — APPROVED** [review]
Wrong formatting

**Pull request merged**